contracts/DxMgnPool.sol

```diff
@@ -129,12 +130,12 @@ contract DxMgnPool is Ownable {
             "Last auction is still running"
         );
         
-        mgnToken.unlockTokens();
-        
         (address sellToken, address buyToken) = buyAndSellToken();
         dx.claimSellerFunds(buyToken, sellToken, address(this), lastParticipatedAuctionIndex);
-        uint amount = dx.balances(address(depositToken), address(this));
-        dx.withdraw(address(depositToken), amount);
+
+        mgnToken.unlockTokens();
```

Comment:
Can you please make this its own PR? It makes it easy to track changes and if necessary revert them individually.